### PR TITLE
Fix IPv6 fragmentation

### LINKED ISSUE #16354: net: ipv6: Zephyr does not reply to fragmented packet
Zephyr receives fragmented packet, but does not reply. The same test is done with Linux DUT and it successfully replies to fragmented echo request.

[ipv6-frag-zephyr.pcap.gz](https://github.com/zephyrproject-rtos/zephyr/files/3212117/ipv6-frag-zephyr.pcap.gz)
[ipv6-frag-linux.pcap.gz](https://github.com/zephyrproject-rtos/zephyr/files/3212119/ipv6-frag-linux.pcap.gz)

### PR DESCRIPTION
**Pull request by jukkar**
Fixes #16354

### CODE REVIEW

File: subsys/net/ip/net_pkt.c [hunk 1]
```diff
@@ -1774,7 +1774,7 @@ int net_pkt_pull(struct net_pkt *pkt, size_t length)
 		c_op->buf->len -= rem;
 		left -= rem;
 		if (left) {
-			memmove(c_op->pos, c_op->pos+rem, rem);
+			memmove(c_op->pos, c_op->pos+rem, left);
```
Comment (tbursztyka) [inline subsys/net/ip/net_pkt.c]: ouch, indeed.

**Review by tbursztyka — APPROVED** [review]

File: subsys/net/ip/net_pkt.c (hunk 1 above)
Comment (pfalcon) [reply subsys/net/ip/net_pkt.c]: > memmove(...)

Gosh, how expensive this is.

Comment (pfalcon) [reply subsys/net/ip/net_pkt.c]: Also, a friendly reminder that regression tests were invented exactly for cases like this...

File: subsys/net/ip/ipv6_fragment.c [hunk 2]
```diff
@@ -317,7 +319,6 @@ static void reassemble_packet(struct net_ipv6_reassembly *reass)
 		goto error;
 	}
 
-
```
Comment (pfalcon) [inline subsys/net/ip/ipv6_fragment.c]: Unrelated change.

**Review by pfalcon — APPROVED** [review]
It's nice to see fixes like this, but we'll never know if we actually fix things, or just munge around, without regression tests added. (Which are hard and boring to write, as we already discussed.)

**Pull request merged**